### PR TITLE
chore(common): iis setup script for dev boxes

### PR DESCRIPTION
**Pull request by mcdurdin**
Relates to #5193.

Helps us developers to configure our development machines more easily.

One day, perhaps we'll have a script to prepare all our local website clones!

See also:

* keymanapp/api.keyman.com#153
* keymanapp/help.keyman.com#411
* keymanapp/keymanweb.com#34
* keymanapp/keyman.com#302

### CODE REVIEW

**Comment by darcywong00** [comment]
drats. the failing Test-14.0 web build is from #5136. I've rerun to see if if the step needs to end with `exit 0`.

File: resources/devbox/iis-https/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+The script must be run as Administrator and can safely be run multiple times.
+
+Once configured, the certificates will work for the following sites (configured in v3.ext):
+- `localhost`
+- `keyman.com.local`
+- `*.keyman.com.local`
+- `keymanweb.com.local`
+- `keymankeyboards.com.local`
+- `keyman.dev.local`
+- `*.keymanweb.com.local`
```
Comment (darcywong00) [inline resources/devbox/iis-https/README.md:16]: I don't think I still use a local version of this, but does tavultesoft.com.local need this too?

**Comment by darcywong00** [comment]
Should this get updated too?
https://github.com/keymanapp/help.keyman.com/blob/master/Readme.md

(and is the docker portion of that still relevant?)

File: resources/devbox/iis-https/README.md (hunk 1 above)
Comment (mcdurdin) [reply resources/devbox/iis-https/README.md:16]: no, it's very much legacy.

**Comment by mcdurdin** [comment]
> Should this get updated too?
> https://github.com/keymanapp/help.keyman.com/blob/master/Readme.md

Probably...

> (and is the docker portion of that still relevant?)

Not at this point

File: resources/devbox/iis-https/server.csr.cnf [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+default_md = sha256
+distinguished_name = dn
+
+[dn]
+C=AU
+ST=Tasmania
+L=Hobart
+O=SIL International
+emailAddress=marc_durdin@sil.org
+CN = keyman.com.local
```
Comment (darcywong00) [inline resources/devbox/iis-https/server.csr.cnf:13]: The setup prompted me for all of these fields, so maybe you can leave your PII off?

File: resources/devbox/iis-https/setup-iis-https.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  echo "This should be unique to your setup."
+  winpty openssl genrsa -des3 -out rootCA.key 2048
+else
+  echo "Skipping, rootCA.key already present"
+fi
+
+# Generate CA certificate
+echo_heading2 "Generating CA certificate"
+if [ ! -f rootCA.pem ]; then
+  winpty openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 3650 -out rootCA.pem
```
Comment (darcywong00) [inline resources/devbox/iis-https/setup-iis-https.sh:40]: Does this script need to be re-run when the certs expire?

**Review by darcywong00 — APPROVED** [review]
I followed the README.md and it was pretty straightforward.

lgtm

File: resources/devbox/iis-https/setup-iis-https.sh (hunk 3 above)
Comment (mcdurdin) [reply resources/devbox/iis-https/setup-iis-https.sh:40]: once every ten years, yep.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.63-alpha](https://keyman.com/downloads/releases/alpha/15.0.63)

**Pull request merged**